### PR TITLE
Index NFT information from Genesis Block

### LINKED ISSUE #1793: Index NFT information from Genesis Block
### Description
To keep track of NFTs, we need to index all possible information available from genesis block.

### Acceptance Criteria

- Index NFT information from genesis block during the start
  - Index following information from NFT substore
    - nftID
    - owner
  - Cache supportedNFTs information 
- Add unit tests if applicable

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #1793 

### How was it solved?

- [x] Index NFT information from genesis block during the start
  - [x] Index following information from NFT substore
    - [x] nftID
    - [x] owner
  - [x] Cache supportedNFTs information
- [x] Add unit tests if applicable (NA)

### How was it tested?
Local

### CODE REVIEW

File: services/blockchain-indexer/shared/indexer/genesisBlock.js [hunk 1]
```diff
@@ -133,9 +137,53 @@ const indexPosModuleAssets = async (dbTrx) => {
...
+		const nftSubstore = (await requestAll(
+			requestConnector,
+			'getGenesisAssetByModule',
+			{ module: MODULE.NFT, subStore: MODULE_SUB_STORE.NFT.NFT_SUB_STORE },
+			nftSubstoreLength,
+		)).NFTSubstore;
+
+		const nftEntries = nftSubstore.map(nftInfo => ({
+			nftID: nftInfo.nftID,
+			owner: nftInfo.owner,
```
Comment (vardan10) [inline services/blockchain-indexer/shared/indexer/genesisBlock.js:153]: Should we also index NFT attributes here?

File: services/blockchain-indexer/shared/database/schema/nft.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+module.exports = {
+	tableName: 'nft',
+	primaryKey: 'nftID',
+	schema: {
+		nftID: { type: 'string' },
+		owner: { type: 'string' },
```
Comment (vardan10) [inline services/blockchain-indexer/shared/database/schema/nft.js:21]: Should we add strict not null check for safety?
```suggestion
		nftID: { type: 'string', null: false },
		owner: { type: 'string', null: false },
```

File: services/blockchain-indexer/shared/indexer/genesisBlock.js [hunk 3]
```diff
@@ -133,9 +137,53 @@ const indexPosModuleAssets = async (dbTrx) => {
...
+		await nftTable.upsert(nftEntries, dbTrx);
+	}
+};
+
+const cacheSupportedNFTs = async (supportedNFTsSubstoreLength) => {
+	if (supportedNFTsSubstoreLength > 0 && !supportedNFTs) {
+		supportedNFTs = (await requestAll(
+			requestConnector,
+			'getGenesisAssetByModule',
+			{ module: MODULE.NFT, subStore: MODULE_SUB_STORE.NFT.NFT_SUB_STORE },
```
Comment (vardan10) [inline services/blockchain-indexer/shared/indexer/genesisBlock.js:165]:
```suggestion
			{ module: MODULE.NFT, subStore: MODULE_SUB_STORE.NFT.SUPPORTED_NFT },
```

File: services/blockchain-indexer/shared/indexer/genesisBlock.js (hunk 1 above)
Comment (sameersubudhi) [reply services/blockchain-indexer/shared/indexer/genesisBlock.js:153]: Doesn't seem necessary.

File: services/blockchain-indexer/shared/database/schema/nft.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ */
+module.exports = {
+	tableName: 'nft',
+	primaryKey: 'nftID',
+	schema: {
+		nftID: { type: 'string', null: false },
+		owner: { type: 'string', null: false },
+	},
+	indexes: {
+		owner: { type: 'key' },
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/database/schema/nft.js:24]:
```suggestion
		owner: { type: 'string', null: false },
		escrowChainID: { type: 'string', null: true },
	},
	indexes: {
		owner: { type: 'key' },
		escrowChainID: { type: 'key' },
```

File: services/blockchain-indexer/shared/indexer/genesisBlock.js [hunk 5]
```diff
@@ -133,9 +137,53 @@ const indexPosModuleAssets = async (dbTrx) => {
...
+const indexNFTSubstore = async (nftSubstoreLength, dbTrx) => {
+	const nftTable = await getNFTTable();
+
+	if (nftSubstoreLength > 0) {
+		const nftSubstore = (await requestAll(
+			requestConnector,
+			'getGenesisAssetByModule',
+			{ module: MODULE.NFT, subStore: MODULE_SUB_STORE.NFT.NFT_SUB_STORE },
+			nftSubstoreLength,
+		)).NFTSubstore;
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/indexer/genesisBlock.js:149]:
```suggestion
		const response = await requestAll(
			requestConnector,
			'getGenesisAssetByModule',
			{ module: MODULE.NFT, subStore: MODULE_SUB_STORE.NFT.NFT_SUB_STORE },
			nftSubstoreLength,
		);
		const nftSubstore = response[MODULE_SUB_STORE.NFT.NFT_SUB_STORE];
```

File: services/blockchain-indexer/shared/indexer/genesisBlock.js [hunk 6]
```diff
@@ -133,9 +137,53 @@ const indexPosModuleAssets = async (dbTrx) => {
...
+
+const indexNFTModuleAssets = async (dbTrx) => {
+	const genesisBlockAssetsLength = await requestConnector('getGenesisAssetsLength', { module: MODULE.NFT });
+	/* eslint-disable max-len */
+	const nftSubstoreLength = genesisBlockAssetsLength[MODULE.NFT][MODULE_SUB_STORE.NFT.NFT_SUB_STORE];
+	const supportedNFTsSubstoreLength = genesisBlockAssetsLength[MODULE.NFT][MODULE_SUB_STORE.NFT.SUPPORTED_NFT];
+	/* eslint-enable max-len */
+
+	await indexNFTSubstore(nftSubstoreLength, dbTrx);
+	// TODO: Remove cache once method to fetch sdupported NFTs exposed from SDK
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/indexer/genesisBlock.js:179]:
```suggestion
	// TODO: Remove cache once method to fetch supported NFTs exposed from SDK
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/blockchain-indexer/shared/database/schema/nft.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+module.exports = {
+	tableName: 'nft',
+	primaryKey: 'nftID',
+	schema: {
+		nftID: { type: 'string', null: false },
+		owner: { type: 'string', null: false },
+		escrowChainID: { type: 'string' },
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/database/schema/nft.js:22]:
```suggestion
		escrowChainID: { type: 'string', null: true },
```

**Review by sameersubudhi — APPROVED** [review]

**Review by vardan10 — APPROVED** [review]

**Pull request merged**